stbt_auto_selftest.py

```diff
@@ -147,6 +147,20 @@ def init_worker():
     signal.signal(signal.SIGINT, signal.SIG_IGN)
 
 
+def iterate_with_progress(iterator, width=20, stream=sys.stderr):
```

Comment:
Nitpick on the naming: an iterator, technically, only supports `next()`, not `len()`. Python's terminology for this seems to be "sequence": https://docs.python.org/2/library/functions.html#len